### ISSUE TITLE
Bug: SC2102 on Associative Array key existence check with repeated characters

### ISSUE THREAD

**Issue by zachriggle** [koalaman/shellcheck#2285]
#### For bugs
- Rule Id SC2102
- My shellcheck version "online"
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/opt/brew/bin/bash
declare -A commands

commands[aa]=3

[[ -v commands[a] ]] && echo a
[[ -v commands[aa] ]] && echo aa
[[ -v commands[aaa] ]] && echo aaa
```

**Output**

```
$ ./myscript
aa
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 7:
[[ -v commands[aa] ]] && echo aa
              ^-- SC2102: Ranges can only match single chars (mentioned due to duplicates).
 
Line 8:
[[ -v commands[aaa] ]] && echo aaa
              ^-- SC2102: Ranges can only match single chars (mentioned due to duplicates).
```

#### Here's what I wanted or expected to see:

No error.

**Comment by candacebutler8879** [comment]
I don't understand I'm just trying to figure out how to get into my email account that got hacked and I'm not as smart as you so can you dumb it down for me please

**Comment by koalaman** [comment]
ShellCheck parsed the arguments to `-v` (and also other dereferencing operators like `-gt` and `-eq`) as regular shell words, causing the `[..]` to be considered a glob range. The warning is now suppressed whenever the word is used as an argument to such an operator. Thanks!

**Comment by LinuxOnTheDesktop** [comment]
Hello. I have Shellcheck 0.9.0 and it seems to me that I am running into the problem. For, the following code produces

>   22:11  warning  shellcheck  Ranges can only match single chars (mentioned due to duplicates). [SC2102]   ​note


```
#!/usr/bin/env bash

declare -r -A sfs_path=(
	[scripts_cloudball_fetch]='whatever'
)

declare -A sfs_path

function isArray_indexed {
	# I have removed the code from this function.
	return 0
}

function ensureSet {
	# I have removed the code from this function.
	return 0
}

function testing {
	ensureSet \
		"${FUNCNAME[0]}" \
		sfs_path[scripts_cloudball_fetch]
	echo "${sfs_path[scripts_cloudball_fetch]}"
}

testing

# EOF
```

Perhaps though I make some mistake. For - for one thing - while the above script does not throw an actual run-time error, the code from which it was simplied does . .
